File: javascript/module1/solutions/solution3.js
function getWeaponDamage(weaponBaseDamage, skillLevel) {
  const skillDamageMultiplier = 0.05;
  const randomFactorMaxValue = 10;
  const minDamage = 20;

  const skillFactor = weaponBaseDamage * (skillLevel * skillDamageMultiplier);
  const randomFactor = Math.random() * randomFactorMaxValue;

  const totalDamage = weaponBaseDamage + skillFactor + randomFactor;

  return Math.max(totalDamage, minDamage);
}

File: csharp/module1/solutions/solution3.cs
public double GetWeaponDamage(double weaponBaseDamage, int skillLevel) {
    const double skillDamageMultiplier = 0.05;
    const int randomFactorMaxValue = 10;
    const int minDamage = 20;

    double skillFactor = weaponBaseDamage * (skillLevel * skillDamageMultiplier);
    Random random = new Random();
    double randomFactor = random.NextDouble() * randomFactorMaxValue;

    double totalDamage = weaponBaseDamage + skillFactor + randomFactor;

    return Math.Max(totalDamage, minDamage);
}
